tests/ops/test_qubit_ops.py

```diff
@@ -865,6 +867,153 @@ def test_controlled_phase_shift_decomp(self, phi):
+        """Tests that the SingleExcitationPlus operation calculates the correct matrix"""
+        op = qml.SingleExcitationMinus(phi, wires=[0, 1])
+        res = op.matrix
+        exp = SingleExcitationMinus(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("phi", [-0.1, 0.2, np.pi/4])
+    def test_single_excitation_minus_generator(self, phi):
+        """Tests that the SingleExcitationPlus operation calculates the correct generator"""
+        op = qml.SingleExcitationMinus(phi, wires=[0, 1])
+        g, a = op.generator
+        res = expm(1j * a * g * phi)
+        exp = SingleExcitationMinus(phi)
+        assert np.allclose(res, exp)
+
+    @pytest.mark.parametrize("excitation", [qml.SingleExcitation, qml.SingleExcitationPlus,
+                                            qml.SingleExcitationMinus])
+    def test_autograd(self, excitation):
+        """Tests that operations are computed correctly using the
+        autograd interface"""
+
+        pytest.importorskip("autograd")
+
+        dev = qml.device('default.qubit.autograd', wires=2)
+        state = np.array([0, -1 / np.sqrt(2), 1 / np.sqrt(2), 0])
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            excitation(phi, wires=[0, 1])
+
+            return qml.state()
+
+        assert np.allclose(state, circuit(np.pi / 2))
+
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.SingleExcitation, -0.1),
+                                                     (qml.SingleExcitationPlus, 0.2),
+                                                     (qml.SingleExcitationMinus, np.pi/4)])
+    def test_autograd_grad(self, excitation, phi):
+        """Tests that gradients are computed correctly using the
+        autograd interface"""
+
+        pytest.importorskip("autograd")
+
+        dev = qml.device('default.qubit.autograd', wires=2)
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            excitation(phi, wires=[0, 1])
+
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(qml.grad(circuit)(phi), np.sin(phi))  # gradient = sin(phi)
+
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.SingleExcitation, -0.1),
+                                                     (qml.SingleExcitationPlus, 0.2),
+                                                     (qml.SingleExcitationMinus, np.pi / 4)])
+    def test_tf(self, excitation, phi):
+        """Tests that gradients and operations are computed correctly using the
+        tensorflow interface"""
+
+        pytest.importorskip("tensorflow")
+
+        dev = qml.device('default.qubit.tf', wires=2)
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            excitation(phi, wires=[0, 1])
+
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(qml.grad(circuit)(phi), np.sin(phi))  # gradient = sin(phi)
```

Comment:
```suggestion
    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
    @pytest.mark.parametrize(("excitation", "phi"), [(qml.SingleExcitation, -0.1),
                                                     (qml.SingleExcitationPlus, 0.2),
                                                     (qml.SingleExcitationMinus, np.pi / 4)])
    def test_tf(self, excitation, phi, diff_method):
        """Tests that gradients and operations are computed correctly using the
        tensorflow interface"""

        tf = pytest.importorskip("tensorflow")

        dev = qml.device('default.qubit', wires=2)

        @qml.qnode(dev, interface="tf", diff_method=diff_method)
        def circuit(phi):
            qml.PauliX(wires=0)
            excitation(phi, wires=[0, 1])
            return qml.expval(qml.PauliZ(0))
            
        phi_t = tf.Variable(phi, dtype=tf.float64)
        
        with tf.GradientTape() as tape:
            res = circuit(phi_t)

        grad = tape.gradient(res, phi_t)
        assert np.allclose(grad, np.sin(phi))
```

Reply:
@ixfoduap I think this isn't actually testing the TF interface/gradient! I've made the following changes above:

- Added a parametrization for the diff method, so that the test checks both the parameter-shift _and_ backprop gradient methods
- Changing `default.qubit.tf` to `default.qubit` (so that both diff methods can be tested)
- Adding `interface="tf"` -- this is the key requirement to enforce TensorFlow usage
- Swapping `qml.grad(..)` (which uses Autograd) for the TensorFlow gradient function.

Reply:
Just in case my suggestion above has any bugs, @antalszava should be able to help out quickly fix it up!